india_compliance/gst_india/doctype/bill_of_entry/bill_of_entry.py

```diff
@@ -0,0 +1,504 @@
+# Copyright (c) 2023, Resilient Tech and contributors
+# For license information, please see license.txt
+
+import json
+
+import frappe
+from frappe import _
+from frappe.model.document import Document
+from frappe.model.mapper import get_mapped_doc
+from frappe.utils import today
+import erpnext
+from erpnext.accounts.general_ledger import make_gl_entries
+from erpnext.controllers.accounts_controller import AccountsController
+
+from india_compliance.gst_india.utils import get_gst_accounts_by_type
+
+
+class BillofEntry(Document):
+    def onload(self):
+        if self.docstatus != 1:
+            return
+
+        existing_journal_entry = frappe.db.get_value(
+            "Journal Entry Account",
+            {
+                "reference_type": "Bill of Entry",
+                "reference_name": self.name,
+                "docstatus": 1,
+            },
+            "name",
+        )
+
+        if existing_journal_entry:
+            self.set_onload("existing_journal_entry", existing_journal_entry)
```

Comment:
[optional]

we aren't really using the values in `existing_journal_entry` or `existing_bill_of_entry`. In such cases, using `frappe.db.exists` coupled with a variable name like `journal_entry_exists` better conveys what we are trying to achieve.